variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/VariantClassifier.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.commons.core.models;
+
+import java.util.regex.Pattern;
+
+/**
+ * Implementation of Variant type classification based on Sequence Ontology (SO) definitions.
+ */
+public class VariantClassifier {
+
+    private static Pattern alphaRegExPattern = Pattern.compile("[A-Z]+");
+    /***
+     * Get the type of a given variant based on Sequence Ontology (SO) definitions.
+     * See https://docs.google.com/spreadsheets/d/1YH8qDBDu7C6tqULJNCrGw8uBjdW3ZT5OjTkJzGNOZ4E/edit#gid=1433496764
+     * @TODO: might need to be merged eventually into {@link uk.ac.ebi.eva.commons.core.models.AbstractVariant#getType()}
+     *
+     * @param reference Reference allele - expects trimmed allele
+     * @param alternate Alternate allele - expects trimmed allele
```

Comment:
what do you mean here? the first thing you do in the function is calling `.trim()` on the alleles